contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -0,0 +1,280 @@
+     * @param anchorPeriod_ The minimum amount of time required for the old uniswap price accumulator to be replaced
+     */
+    constructor(OpenOraclePriceData priceData_,
+                address reporter_,
+                uint anchorToleranceMantissa_,
+                uint anchorPeriod_,
+                TokenConfig[] memory configs) UniswapConfig(configs) public {
+        priceData = priceData_;
+        reporter = reporter_;
+        anchorPeriod = anchorPeriod_;
+
+        require(anchorToleranceMantissa_ < 100e16, "anchor tolerance is too high");
+        upperBoundAnchorRatio = 100e16 + anchorToleranceMantissa_;
+        lowerBoundAnchorRatio = 100e16 - anchorToleranceMantissa_;
+
+        for (uint i = 0; i < configs.length; i++) {
+            TokenConfig memory config = configs[i];
+            address uniswapMarket = config.uniswapMarket;
+            if (config.priceSource == PriceSource.REPORTER) {
+                require(uniswapMarket != address(0), "reported prices must have an anchor");
+                uint cumulativePrice = currentCumulativePrice(config);
+                oldObservations[uniswapMarket].timestamp = block.timestamp;
+                newObservations[uniswapMarket].timestamp = block.timestamp;
+                oldObservations[uniswapMarket].acc = cumulativePrice;
+                newObservations[uniswapMarket].acc = cumulativePrice;
+            } else {
+                require(uniswapMarket == address(0), "only reported prices utilize an anchor");
+            }
+        }
+    }
+
+    /**
+     * @notice Get the official price for a symbol
+     * @param symbol The symbol to fetch the price of
+     * @return Price denominated in USD, with 6 decimals
+     */
+    function price(string memory symbol) public view returns (uint) {
+        TokenConfig memory config = getTokenConfigBySymbol(symbol);
+        return priceInternal(config);
+    }
+
+    function priceInternal(TokenConfig memory config) internal view returns (uint) {
+        if (config.priceSource == PriceSource.REPORTER) return prices[config.symbolHash];
+        if (config.priceSource == PriceSource.FIXED_USD) return config.fixedPrice;
+        if (config.priceSource == PriceSource.FIXED_ETH) {
+            uint usdPerEth = prices[ethHash];
+            require(usdPerEth > 0, "ETH price not set, cannot convert to dollars");
+            return mul(usdPerEth, config.fixedPrice) / config.baseUnit;
+        }
+    }
+
+    /**
+     * @notice Get the underlying price of a cToken
+     * @dev Implements the PriceOracle interface for Compound v2.
+     * @param cToken The cToken address for price retrieval
+     * @return The price for the given cToken address
+     */
+    function getUnderlyingPrice(address cToken) public view returns (uint) {
+        TokenConfig memory config = getTokenConfigByCToken(cToken);
+        return mul(1e30, priceInternal(config)) / config.baseUnit;
```

Comment:
FWIW, I think we should move all "magic constants" to be constants-- otherwise it's pretty difficult to understand where these come from.

Reply:
addressed #102